src/menu/views/load_disk.c

```diff
@@ -163,6 +166,8 @@ void view_load_disk_init (menu_t *menu) {
     if (err != DISK_OK) {
         menu_show_error(menu, convert_error_message(err));
     }
+
+    boxart = component_boxart_init_dd(menu->storage_prefix, menu->load.disk_info.id);
```

Comment:
You're allocating memory for boxart but never free it. Use `component_boxart_free` function - example is in the ROM load view.